Source/WebKit/WebProcess/GPU/graphics/RemoteImageBufferProxy.h

```diff
@@ -254,6 +260,7 @@ class RemoteImageBufferProxy : public WebCore::ConcreteImageBuffer<BackendType>
         ASSERT(resolutionScale() == 1);
         auto& mutableThis = const_cast<RemoteImageBufferProxy&>(*this);
         mutableThis.flushDrawingContextAsync();
+        prepareForBackingStoreChange();
         m_remoteRenderingBackendProxy->putPixelBufferForImageBuffer(m_renderingResourceIdentifier, pixelBuffer, srcRect, destPoint, destFormat);
         setNeedsFlush(true);
```

Comment:
This call to `setNeedsFlush(true)` will make sense if we leave `setNeedsFlush()` unchanged. With this patch `prepareForBackingStoreChange()` may be called twice: one before calling `putPixelBufferForImageBuffer()` and the second from `setNeedsFlush()`.